libsolidity/interface/StandardCompiler.cpp

```diff
@@ -100,20 +100,21 @@ Json::Value formatErrorWithException(
 	return formatError(_warning, _type, _component, message, formattedMessage, sourceLocation);
 }
 
-set<string> requestedContractNames(Json::Value const& _outputSelection)
+map<string, set<string>> requestedContractNames(Json::Value const& _outputSelection)
 {
-	set<string> names;
+	map<string, set<string>> contracts;
 	for (auto const& sourceName: _outputSelection.getMemberNames())
 	{
+		string key = (sourceName == "*") ? "" : sourceName;
 		for (auto const& contractName: _outputSelection[sourceName].getMemberNames())
 		{
-			/// Consider the "all sources" shortcuts as requesting everything.
-			if (contractName == "*" || contractName == "")
-				return set<string>();
-			names.insert((sourceName == "*" ? "" : sourceName) + ":" + contractName);
+			string value = (contractName == "*") ? "" : contractName;
+			if (key == "" && contractName == "")
```

Comment:
Do we still have to handle this as a special case here?

Reply:
It was just a cut, but should work without